rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java

```diff
@@ -166,4 +189,70 @@ private Block getByJsonBlockId(String id) {
             }
         }
     }
+
+    private List<TransactionTrace> buildBlockTraces(Block block) {
+        return buildBlockTraces(block, null);
+    }
+
+    private List<TransactionTrace> buildBlockTraces(Block block, TraceFilterRequest traceFilterRequest) {
+        List<TransactionTrace> blockTraces = new ArrayList<>();
+
+        if (block != null && block.getNumber() != 0) {
+            List<Transaction> txList = block.getTransactionsList();
+
+            ProgramTraceProcessor programTraceProcessor = new ProgramTraceProcessor();
+            Block parent = this.blockchain.getBlockByHash(block.getParentHash().getBytes());
+            this.blockExecutor.traceBlock(programTraceProcessor, VmConfig.LIGHT_TRACE, block, parent.getHeader(), false, false);
+
+            if (traceFilterRequest != null) {
+                Stream<Transaction> txStream = block.getTransactionsList().stream();
+
+                if (traceFilterRequest.getFromAddress() != null && !traceFilterRequest.getFromAddress().isEmpty()) {
+                    List<RskAddress> addresses = traceFilterRequest.getFromAddressAsRskAddresses();
+                    txStream = txStream.filter(tx -> tx.getSender().getBytes().length > 0 && addresses.contains(tx.getSender()));
+                }
+
+                if (traceFilterRequest.getToAddress() != null && !traceFilterRequest.getToAddress().isEmpty()) {
+                    List<RskAddress> addresses = traceFilterRequest.getToAddressAsRskAddresses();
+                    txStream = txStream.filter(tx -> tx.getReceiveAddress().getBytes().length > 0 && addresses.contains(tx.getReceiveAddress()));
+                }
+
+                txList = txStream.collect(Collectors.toList());
+            }
+
+            for (Transaction tx : txList) {
+                TransactionInfo txInfo = receiptStore.getInMainChain(tx.getHash().getBytes(), this.blockStore).orElse(null);
+                Objects.requireNonNull(txInfo);
+                txInfo.setTransaction(tx);
+
+                SummarizedProgramTrace programTrace = (SummarizedProgramTrace) programTraceProcessor.getProgramTrace(tx.getHash());
+
+                if (programTrace == null) {
+                    blockTraces.clear();
+                    return null;
+                }
+
+                List<co.rsk.rpc.modules.trace.TransactionTrace> traces = TraceTransformer.toTraces(programTrace, txInfo, block.getNumber());
+
+                blockTraces.addAll(traces);
+            }
+        }
+
+        return blockTraces;
+    }
+
+    private Block getBlockByTagOrNumber(String strBlock, BigInteger biBlock) {
+        if (strBlock.equalsIgnoreCase(LATEST_BLOCK)) {
+            return this.blockchain.getBestBlock();
+        } else if (strBlock.equalsIgnoreCase(EARLIEST_BLOCK)) {
+            return this.blockchain.getBlockByNumber(0);
+        } else if (strBlock.equalsIgnoreCase(PENDING_BLOCK)) {
+            Optional<Block> latestBlock = minerServer.getLatestBlock();
```

Comment:
it may be not enough to call `minerServer.getLatestBlock()` to get a pending block. Check [this](https://github.com/rsksmart/rskj/blob/91e5f60de29160ca8b96542bd0e59b0a57c31f13/rskj-core/src/main/java/co/rsk/rpc/ExecutionBlockRetriever.java#L71-L93) example out.

Reply:
Done